src/operator/controllers/intents/reconciler.go

```diff
@@ -0,0 +1,336 @@
+				Namespace: clientIntents.Namespace,
+			},
+		}
+		requests = append(requests, request)
+	}
+
+	return requests
+}
+
+func (r *Reconciler) mapPGServerConfToClientIntents(ctx context.Context, obj client.Object) []reconcile.Request {
+	pgServerConf := obj.(*otterizev1alpha3.PostgreSQLServerConfig)
+	logrus.Infof("Enqueueing client intents for PostgreSQLServerConfig change %s", pgServerConf.Name)
+
+	intentsToReconcile := make([]otterizev1alpha3.ClientIntents, 0)
+	intentsList := otterizev1alpha3.ClientIntentsList{}
+	dbInstanceName := pgServerConf.Name
+	err := r.client.List(ctx,
+		&intentsList,
+		&client.MatchingFields{otterizev1alpha3.OtterizeTargetServerIndexField: dbInstanceName},
+	)
+	if err != nil {
+		logrus.Errorf("Failed to list client intents targeting %s: %v", dbInstanceName, err)
+	}
+	intentsToReconcile = append(intentsToReconcile, intentsList.Items...)
+
+	requests := make([]reconcile.Request, 0)
+	for _, clientIntents := range intentsToReconcile {
+		requests = append(requests, reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name:      clientIntents.Name,
+				Namespace: clientIntents.Namespace,
+			},
+		})
+	}
+	return requests
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	var intents otterizev1alpha3.ClientIntents
+	err := r.client.Get(ctx, req.NamespacedName, &intents)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, errors.Wrap(err)
+	}
+
+	dbNames := extractDBInstanceNames(intents)
+	if len(dbNames) == 0 {
+		return ctrl.Result{}, nil
+	}
+
+	for _, databaseName := range dbNames {
+		pgServerConfigs := otterizev1alpha3.PostgreSQLServerConfigList{}
+		err := r.client.List(ctx, &pgServerConfigs)
+		if err != nil {
+			r.recorder.Eventf(&intents, v1.EventTypeWarning, ReasonErrorFetchingPostgresServerConfig,
+				"Error trying to fetch '%s' PostgresServerConf for client '%s'. Error: %s",
+				databaseName, intents.GetServiceName(), err.Error())
+			return ctrl.Result{}, errors.Wrap(err)
+		}
+		pgServerConf, err := findMatchingPGServerConfForDBInstance(databaseName, pgServerConfigs)
+		if err != nil {
+			r.recorder.Eventf(&intents, v1.EventTypeWarning, ReasonMissingPostgresServerConfig,
+				"Could not find matching PostgreSQLServerConfig. Error: %s", err.Error())
+			return ctrl.Result{}, nil // Not returning error on purpose, missing PGServerConf - record event and move on
+		}
+
+		pgConfigurator := databaseconfigurator.NewPostgresConfigurator(pgServerConf.Spec, r.client)
+		connectionString := pgConfigurator.FormatConnectionString(pgServerConf.Spec.DatabaseName)
+		conn, err := pgx.Connect(ctx, connectionString)
+		if err != nil {
+			pgErr, ok := databaseutils.TranslatePostgresConnectionError(err)
+			if ok {
+				return ctrl.Result{}, errors.Wrap(fmt.Errorf(pgErr))
+			}
+			return ctrl.Result{}, errors.Wrap(err)
+		}
+		pgConfigurator.SetConnection(ctx, conn)
```

Comment:
There's some code duplication between this and the related reconciler in the intents-operator. Can we share the code? I'm picturing something like `databaseconfigurator.NewFromPGServerConf(ctx, pgServerConf, r.client)...`